### PR TITLE
Fix S3 third party upload handling (gravitational#3322)

### LINKED ISSUE #3322: handling third party s3 session upload - context needs to be aware of the SSE
**What happened**: During testing for #3054 the upload for a session on third party S3 must not set the ServerSideEncryption, which is only relevant for handling AWS S3. This would otherwise end in an error:

```
WARN [AUDIT]     "Session upload failed: \nERROR REPORT:\nOriginal Error: *trace.BadParameterError InvalidDigest: The Content-MD5 you specified is not valid.\n\tstatus code: 400, request id: ..., host id: ...
```

**What you expected to happen**: Teleport uploads a session when disabling SSE.

**How to reproduce it (as minimally and precisely as possible)**: Using a third party s3 backend and setting disabling the SSE directive:

```
teleport:
  storage:
    audit_sessions_uri: s3://<bucket-name>/sessions?endpoint=<endpoint>&disablesse=1
```

**Environment**:
- Teleport version (use `teleport version`): v4.2.2
- Tsh version (use `tsh version`): v4.2.2
- OS (e.g. from /etc/os-release): RHEL 7.7

### PR DESCRIPTION
**Pull request by balpert89**
This PR fixes #3322. ServerSideEncryption is only used with KMS on AWS (as far as I am aware of) and needs to be not set during third party scenarios.

### CODE REVIEW

**Comment by webvictim** [comment]
ok to test

**Review by fspmarshall — COMMENTED** [review]
Code looks good.  As far as I know this change is inkeeping with the intent of the `DisableServerSideEncryption` option, but I'm definitely not an AWS expert, so I defer to @webvictim and @klizhentas on this one.

File: lib/events/s3sessions/s3handler.go [hunk 1]
```diff
@@ -166,13 +166,22 @@ func (l *Handler) Close() error {
...
 func (l *Handler) Upload(ctx context.Context, sessionID session.ID, reader io.Reader) (string, error) {
+	var err error
 	path := l.path(sessionID)
-	_, err := l.uploader.UploadWithContext(ctx, &s3manager.UploadInput{
-		Bucket:               aws.String(l.Bucket),
-		Key:                  aws.String(path),
-		Body:                 reader,
-		ServerSideEncryption: aws.String(s3.ServerSideEncryptionAwsKms),
-	})
+	if l.Config.DisableServerSideEncryption {
```
Comment (russjones) [inline lib/events/s3sessions/s3handler.go]: Can you change this logic to:

```go
uploadInput := &s3manager.UploadInput{
    Bucket:       aws.String(l.Bucket),
    Key:          aws.String(path),
    Body:         reader,
})
if l.Config.DisableServerSideEncryption == false {
    uploadInput.ServerSideEncryption = aws.String(s3.ServerSideEncryptionAwsKms)
}
_, err := l.uploader.UploadWithContext(ctx, uploadInput)
```

Comment (balpert89) [reply lib/events/s3sessions/s3handler.go]: Done :)

Comment (russjones) [reply lib/events/s3sessions/s3handler.go]: You need either a `!` or `== false` in the `if`statement.

**Review by russjones — APPROVED** [review]

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Pull request merged**